src/pbft/pbft.h

```diff
@@ -129,25 +135,54 @@ namespace pbft
       mem = (char*)malloc(mem_size);
       bzero(mem, mem_size);
 
-      auto exec_command = ([](
-                             Byz_req* inb,
-                             Byz_rep* outb,
-                             _Byz_buffer* non_det,
-                             int client,
-                             bool ro,
-                             Seqno n) { return 0; });
-
       pbft_network = std::make_unique<PbftEnclaveNetwork>(local_id, channels);
+      pbft_config = std::make_unique<PbftConfigCCF>();
 
-      Byz_init_replica(
+      auto used_bytes = Byz_init_replica(
         node_info,
         mem,
         mem_size,
-        exec_command,
+        pbft_config->get_exec_command(),
         0,
         0,
         pbft_network.get(),
         &message_receiver_base);
+
+      service_mem = mem + used_bytes;
+
+      static std::unique_ptr<ClientProxy<uint64_t, void>> client_proxy;
+      LOG_INFO << "Setting up client proxy " << std::endl;
+      client_proxy.reset(
+        new ClientProxy<uint64_t, void>(*message_receiver_base));
+
+      auto cb = [](Reply* m, void* ctx) {
+        auto cp = (ClientProxy<uint64_t, void>*)ctx;
+        cp->recv_reply(m);
+      };
+
+      message_receiver_base->register_reply_handler(cb, client_proxy.get());
+
+      callbacks[pbft::Callbacks::ON_REQUEST] =
+        [&](kv::TxHistory::CallbackArgs args) {
+          auto caller = std::get<0>(args.id);
+          auto session = std::get<1>(args.id);
+          auto version = std::get<2>(args.id);
```

Comment:
As per my previous comment, this cannot be `version` at this point since this is called before the execution of the request.